pynn_spinnaker/random.py

```diff
@@ -59,6 +120,16 @@ def _estimate_dist_max_value(self, distribution, parameters):
         else:
             return self._dist_estimate_max_value[distribution](parameters)
 
+    def _check_dist_parameters(self, distribution, parameters):
+        if not self._supports_dist(distribution):
+            raise NotImplementedError("SpiNNaker native RNG does not support"
+                                      "%s distributions" % distribution)
+        else:
+            valid_parameters = (True, "")
+            if distribution in self._dist_check_parameters.keys():
```

Comment:
I think, at least in Python 2.7, calling ``keys`` is bad as it copies the keys out of the dictionary into a list - just ``if distribution in self._dist_check_parameters`` does the same.